recipes/go-dep/meta.yaml

```diff
@@ -0,0 +1,45 @@
+{% set goname = "github.com/golang/dep" %}
+{% set version = "0.5.0" %}
+
+{% set name = goname.split('/')[-1] %}
+{% set pkg_src = ('src/'+goname).replace("/", "\\" if win else "/") %}
+
+package:
+  name: go-{{ name|lower }}
+  version: {{ version }}
+
+source:
+  - folder:  {{ pkg_src }}
+    url: https://{{ goname }}/archive/v{{ version }}.tar.gz  # [unix]
+    sha256: 65c31c8d51513dfabd25f1f37ecd22a38ab3bcdd5893bdf10f54941250e4f19d  # [unix]
+    git_url: https://{{ goname }}  # [win]
+    git_tag: v{{ version }}  # [win]
```

Comment:
What's the reason for using different sources on difference platforms?

Reply:
Interesting. Yeah the Windows symlink story has been weak historically. Looks like `libarchive` doesn't really have a solution for that.

That said, we might be able to solve it with [this patch]( https://github.com/AnacondaRecipes/libarchive-feedstock/blob/99d01d0f5a4c197e1c807e3e020168bc92f5c557/recipe/0010-Use-CreateSymbolicLinkW-on-Windows-for-symlinks-WIP.patch ), which is being worked on in PR ( https://github.com/libarchive/libarchive/pull/1030 ). Could you please raise an issue about this on the conda-forge `libarchive` feedstock?